src/glob.rs

```diff
@@ -528,44 +669,272 @@ mod test {
         ComparableFilePatterns {
           base: "other".to_string(),
           include: Some(vec!["other/**/*.js".to_string()]),
-          exclude: vec![],
+          exclude: vec!["other/**/*.ts".to_string()],
         }
       ]
     );
   }
 
+  #[track_caller]
+  fn run_file_patterns_match_test(
+    file_patterns: &FilePatterns,
+    path: &Path,
+    kind: PathKind,
+    expected: FilePatternsMatch,
+  ) {
+    assert_eq!(
+      file_patterns.matches_path_detail(path, kind),
+      expected,
+      "path: {:?}, kind: {:?}",
+      path,
+      kind
+    );
+    assert_eq!(
+      file_patterns.matches_path(path, kind),
+      match expected {
+        FilePatternsMatch::Passed
+        | FilePatternsMatch::PassedOptedOutExclude => true,
+        FilePatternsMatch::Excluded => false,
+      }
+    )
+  }
+
+  #[test]
+  fn file_patterns_include() {
+    let cwd = std::env::current_dir().unwrap();
+    // include is a closed set
+    let file_patterns = FilePatterns {
+      base: cwd.clone(),
+      include: Some(PathOrPatternSet(vec![
+        PathOrPattern::from_relative(&cwd, "target").unwrap(),
+        PathOrPattern::from_relative(&cwd, "other/**/*.ts").unwrap(),
+      ])),
+      exclude: PathOrPatternSet(vec![]),
+    };
+    let run_test =
+      |path: &Path, kind: PathKind, expected: FilePatternsMatch| {
+        run_file_patterns_match_test(&file_patterns, path, kind, expected);
+      };
+    run_test(&cwd, PathKind::Directory, FilePatternsMatch::Passed);
```

Comment:
Not sure I understand that - if we didn't specify cwd as included why is it matched?

Reply:
It's "passed" because the directory contents might have matches. The directory matching behaviour is slightly different. I moved this behaviour down from the CLI.

Reply:
I see, thanks for explanation